### PR TITLE
Allow orgs with leading numbers in name in GitRemote

### PR DESCRIPTION
**Pull request by timtebeek**
## What's your motivation?
- Fixes https://github.com/openrewrite/rewrite/issues/4726

## Anyone you would like to review specifically?
@tylervangorder-8451

### CODE REVIEW

**Review by tylervangorder-8451 — COMMENTED** [review]
Thanks for the quick turnaround

**Comment by timtebeek** [comment]
You're welcome! You did _just_ miss the release we did today, but I imagine you're comfortable running snapshot versions if needed. All the best!

File: rewrite-core/src/main/java/org/openrewrite/GitRemote.java [hunk 1]
```diff
@@ -279,7 +279,7 @@ private String repositoryPath(RemoteServerMatch match, URI normalizedUri) {
                     .replaceFirst("^/", "");
         }
 
-        private static final Pattern PORT_PATTERN = Pattern.compile(":\\d+");
+        private static final Pattern PORT_PATTERN = Pattern.compile(":\\d+[^-\\w]"); // Allow orgs with numbers in name
```
Comment (bryceatmoderne) [inline rewrite-core/src/main/java/org/openrewrite/GitRemote.java:282]: github at least supports org names with only numbers.

Comment (timtebeek) [reply rewrite-core/src/main/java/org/openrewrite/GitRemote.java:282]: Guess those will still get the exception thrown then, but at least this works for partial numeric org names. Welcome suggestions for a complete fix!

Comment (bryceatmoderne) [reply rewrite-core/src/main/java/org/openrewrite/GitRemote.java:282]: > Guess those will still get the exception thrown then, but at least this works for partial numeric org names. Welcome suggestions for a complete fix!

I had opened this previously https://github.com/openrewrite/rewrite/pull/4651

Comment (timtebeek) [reply rewrite-core/src/main/java/org/openrewrite/GitRemote.java:282]: Ah, I had no idea. Approved just now & sorry about the conflict. Do let me know if there's ever something open for too long.

**Pull request merged**